charts/kargo/templates/external-webhooks-server/cluster-role.yaml

```diff
@@ -7,6 +7,22 @@ metadata:
     {{- include "kargo.labels" . | nindent 4 }}
     {{- include "kargo.externalWebhooksServer.labels" . | nindent 4 }}
 rules:
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs:
+  - get
+  - list
+  - watch
```

Comment:
This is ok for now, but we'll need to have a follow-up.

Users can be very picky (and rightfully so) about a component that receives traffic from the open internet (as this one will) having such broad access to secrets.

This is precisely why the management controller dynamically expands and contracts API server access to Secrets as Projects are created and deleted.

If you could open a new issue to track this and assign it to me, I'll take care of it.

Reply:
https://github.com/akuity/kargo/issues/4236#issue-3089562624